api/src/main/java/io/druid/data/input/impl/TimestampSpec.java

```diff
@@ -40,6 +40,10 @@
   // this value should never be set for production data
   private final DateTime missingValue;
 
+  // remember last value parsed
+  private Object lastTimeObject = null;
+  private DateTime lastDateTime = null;
```

Comment:
we could possibly make it threadlocal, which should be fine as long as the number of ingestion threads stay small.